src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  useCallback, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { useTodo } from '../../hooks/useTodo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { deleteTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  loading?: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo, loading }) => {
+  const {
+    todos,
+    setTodos,
+    setIsChecked,
+    setErrorMessage,
+    isProcessing,
+    setIsProcessing,
+  } = useTodo();
+  const [title, setTitle] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(false);
+  const [focus, setFocus] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (focus && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [focus]);
+
+  const checkTodo = (todoId: number): void => {
+    const updatedTodos = todos.map(item => (
+      item.id === todoId
+        ? { ...item, completed: !item.completed }
+        : item));
+
+    setTodos(updatedTodos);
+    setIsChecked(updatedTodos.every(item => item.completed));
+  };
+
+  const handleDoubleClick = useCallback(() => {
+    setIsEditing(true);
+    setFocus(true);
+  }, [isEditing]);
```

Comment:
try to fix such warnings 
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/70009205/cc11dee6-033f-471a-be93-bdf0e30602ef)

Reply:
not fixed